src/TreeView/TreeView.test.tsx

```diff
@@ -1121,4 +1121,80 @@ describe('Asyncronous loading', () => {
       expect(firstChild).toHaveFocus()
     })
   })
+
+  it('moves focus to parent item after closing error dialog', async () => {
+    function TestTree() {
+      const [error, setError] = React.useState('Test error')
+
+      return (
+        <TreeView aria-label="Test tree">
+          <TreeView.Item defaultExpanded>
+            Parent
+            <TreeView.SubTree>
+              {error ? (
+                <TreeView.ErrorDialog
+                  onRetry={() => {
+                    setError('')
+                  }}
+                  onDismiss={() => setError('')}
+                >
+                  {error}
+                </TreeView.ErrorDialog>
+              ) : null}
+            </TreeView.SubTree>
+          </TreeView.Item>
+        </TreeView>
+      )
+    }
+
+    const {getByRole} = renderWithTheme(<TestTree />)
+
+    const dialog = getByRole('alertdialog')
+    const parentItem = getByRole('treeitem', {name: 'Parent'})
+
+    // Parent item should not be focused
+    expect(parentItem).not.toHaveFocus()
+
+    // Dialog should be visible
+    expect(dialog).toBeVisible()
+
+    // Press esc to close error dialog
+    fireEvent.keyDown(document.activeElement || document.body, {key: 'Escape'})
+
+    setTimeout(() => {
```

Comment:
The `setTimeout` block here might not get be getting executed. Pulled down the PR real quick to try it out and changed it to:

```js
setTimeout(() => {
  throw Error('test')

  // Parent item should be focused
  expect(parentItem).toHaveFocus()

  // Dialog should not be visible
  expect(dialog).not.toBeVisible()
})
```

And didn't see the test error out. Let me know if you're seeing the same 🤔 Super weird.

Would it make sense to set `jest.useFakeTimers()` at the top-level and then use `jest.runAllTimers()` to force the timers to run and then make the assertions? Something like:

```js
jest.runAllTimers()

// Parent item should be focused
expect(parentItem).toHaveFocus()

// Dialog should not be visible
expect(dialog).not.toBeVisible()
```

Reply:
Great catch! Fixed with a `waitFor` call 👍